pepper-apis/src/main/java/org/broadinstitute/ddp/customexport/export/CustomExporter.java

```diff
@@ -47,20 +56,67 @@
+            throws IOException {
+        RestHighLevelClient elasticsearchClient = ElasticsearchServiceUtil.getElasticsearchClient(this.mainConfig);
+        return createCustomParticipantsFrom(ExportUtil.extractParticipantDataSetByIds(handle, studyDto, userIds, emailStore), handle,
+                studyDto, elasticsearchClient);
+    }
+
+    private static List<CustomExportParticipant> createCustomParticipantsFrom(List<Participant> participants, Handle handle,
+                                                                              StudyDto studyDto,
+                                                                              RestHighLevelClient esClient) {
+        List<CustomExportParticipant> customExportParticipants = new ArrayList<>();
+        for (Participant p : participants) {
+            String familyId = getFamilyId(handle, studyDto, p.getUser().getGuid(), esClient);
+            customExportParticipants.add(new CustomExportParticipant(familyId, p));
+        }
+        return customExportParticipants;
+    }
+
+    private static String getFamilyId(Handle handle, StudyDto studyDto, String userGuid, RestHighLevelClient esClient) throws DDPException {
+        String esIndex = ElasticsearchServiceUtil.getIndexForStudy(handle, studyDto, ElasticSearchIndexType.PARTICIPANTS_STRUCTURED);
+        GetRequest getRequest = new GetRequest(esIndex, "_doc", userGuid);
+        String[] includes = {"dsm"};
+        getRequest.fetchSourceContext(new FetchSourceContext(true, includes, null));
+        GetResponse esResponse;
+        try {
+            esResponse = esClient.get(getRequest, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            throw new DDPException("Error getting ElasticSearch client to use to get family ID for user guid " + userGuid, e);
+        }
+        Map<String, Object> source = esResponse.getSource();
 
-    static List<Participant> extractParticipantDataSetByIds(Handle handle, StudyDto studyDto, Set<Long> userIds) {
-        return ExportUtil.extractParticipantDataSetByIds(handle, studyDto, userIds, emailStore);
+        if (source == null) {
+            throw new DDPException("Error reading ElasticSearch source: source is null for user guid " + userGuid);
+        }
+
+        Object sourceElement = source.get("dsm");
+        if (sourceElement == null) {
+            throw new DDPException("Error reading ElasticSearch source: DSM element is null for user guid" + userGuid);
+        } else if (!(sourceElement instanceof Map)) {
+            throw new DDPException("Error reading ElasticSearch source: DSM element is not a map for user guid " + userGuid);
+        }
+
+        Object familyId = ((Map) sourceElement).get(familyIdHeader);
+        if (familyId == null) {
+            throw new DDPException("Error retrieving family ID from ElasticSearch: familyId is not present for user guid " + userGuid);
```

Comment:
I'm a bit worried about the timing of this. How often will we encounter situation where we need to do a seqr export but DSM hasn't populated the family_id in ES yet? If it's often, will we see a lot of alerts in Slack about this? Will that be too much noise?

Reply:
I typed up a whole set of options and then I realized probably the first thing I should do is find out how often DSM is populating the family_ids.  I'll check that!

Reply:
I talked to Simone and this shouldn't be an issue.  The family ID will be generated when the participant signs up and then immediately added to ElasticSearch.  This export only runs on participants who have completed their survey, at which point they should have family IDs.

Reply:
Great!